### PR TITLE
feat(files): dropzone for file uploads

### PR DESCRIPTION
**Pull request by josephmcg**
**What this PR does** 📖
- refactor files upload to accommodate file drag & drop
- improve tsconfig to support drag & drop event files
- prevent upload if currently in progress
- minor readability/type improvements along the way

**Which issue(s) this PR fixes** 🔨
AP-925

**Special notes for reviewers** 🗒️
- I feel like the way I accessed the ref of a child component is wonky, any better ways?

**Additional comments** 🎤
- discussed with Phill. We eventually want to drag and drop files to move them around / upload to a specific folder on drop. This is a good first step

### CODE REVIEW

File: store/ui/getters.ts [hunk 1]
```diff
@@ -13,6 +13,9 @@ const getters = {
   swiperSlideIndex: (state: UIState) => {
     return state.swiperSlideIndex
   },
+  getFilesIndexLoading: (state: UIState) => {
+    return state.isLoadingFileIndex
+  },
```
Comment (molimauro) [inline store/ui/getters.ts:18]: Its just my idea but im not too much fan of this getters where just a piece of state is returned w/o any sort of transformation. I prefer this way directly in the component:
```
mapState({
    isLoadingFileIndex: state => state.isLoadingFileIndex,
})
```
However its just my opinion we have other several placed where we follow this pattern :)

File: layouts/files.vue [hunk 2]
```diff
@@ -125,10 +129,31 @@ export default Vue.extend({
...
+        return
       }
       this.$store.commit('ui/showSidebar', true)
     },
+    /**
+     * @method handleDrop
+     * @description Allows the drag and drop of files into the filesystem
+     * @param e Drop event data object
+     */
+    async handleDrop(e: DragEvent) {
```
Comment (molimauro) [inline layouts/files.vue:142]: the async can be removed

File: store/ui/getters.ts (hunk 1 above)
Comment (josephmcg) [reply store/ui/getters.ts:18]: Good to know that's an option, thanks! I'm checking this state in many other components too, and was planning on refactoring those to use the getter after this goes through

File: layouts/files.vue (hunk 2 above)
Comment (josephmcg) [reply layouts/files.vue:142]: good call, leftover from copy paste lol

File: store/ui/getters.ts (hunk 1 above)
Comment (molimauro) [reply store/ui/getters.ts:18]: That sounds awesome! Thanks 🙏

**Review by molimauro — APPROVED** [review]

**Comment by stavares843** [comment]
minor comments


- when creating a folder, if i drag a file while folder is being created appears warning saying ''upload in progress, please try again later'' - but that error is incorrect, because is not a upload thats in progress, is a folder thats being created

- i noticed on one of the uploads inside a folder, displayed this message, do we need to show this to the user?

Captura de ecrã 2022-04-13, às 18 48 22

**Review by stavares843 — APPROVED** [review]

**Comment by stavares843** [comment]
[1402](https://satellite-im.atlassian.net/browse/AP-1402)
[1403](https://satellite-im.atlassian.net/browse/AP-1403)

**Pull request merged**